shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.osv import expression
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.addons.shopfloor.utils import to_float
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    @property
+    def dimension_to_update(self):
+        if not hasattr(self.work, "dimension_to_update"):
+            self.work.dimension_to_update = {}
+        return self.work.dimension_to_update
+
+    def _before_response_for_set_quantity(self, picking, line):
+        """Show the packaging dimension screen before the set quantity screen."""
+        if not self.work.menu.set_packaging_dimension:
+            return self._response_for_set_quantity(picking, line)
+        packaging = self._get_next_packaging_to_set_dimension(line.product_id)
+        if packaging:
+            return self._response_for_set_packaging_dimension(picking, line, packaging)
+        return self._response_for_set_quantity(picking, line)
+
+    def _get_domain_packaging_needs_dimension(self):
+        return [
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            "|",
+            ("packaging_length", "=", 0),
+            ("packaging_length", "=", False),
+            ("width", "=", 0),
+            ("width", "=", False),
+            ("height", "=", 0),
+            ("height", "=", False),
+            ("max_weight", "=", 0),
+            ("max_weight", "=", False),
+            ("qty", "=", 0),
+            ("qty", "=", False),
+            ("barcode", "=", False),
+        ]
```

Comment:
```suggestion
        return expression.OR([
            [("packaging_length", "=", 0)],
            [("packaging_length", "=", False)],
            [("width", "=", 0)],
            [("width", "=", False)],
            [("height", "=", 0)],
            [("height", "=", False)],
            [("max_weight", "=", 0)],
            [("max_weight", "=", False)],
            [("qty", "=", 0)],
            [("qty", "=", False)],
            [("barcode", "=", False)],
        ])
```
I think this is a little bit easier to read

Reply:
Therefor you need to import
`from odoo.osv import expression`

Reply:
True, updated.